components/webui/client/src/pages/SearchPage/SearchControls/SearchButton/SubmitButton/index.tsx

```diff
@@ -24,26 +32,74 @@ import styles from "./index.module.css";
  */
 const SubmitButton = () => {
     const searchUiState = useSearchStore((state) => state.searchUiState);
+    const timeRangeOption = useSearchStore((state) => state.timeRangeOption);
     const timeRange = useSearchStore((state) => state.timeRange);
-    const queryIsCaseSensitive = useSearchStore((state) => state.queryIsCaseSensitive);
+    const queryIsCaseSensitive = useSearchStore(
+        (state) => state.queryIsCaseSensitive,
+    );
     const queryString = useSearchStore((state) => state.queryString);
     const selectDataset = useSearchStore((state) => state.selectDataset);
-    const updateCachedDataset = useSearchStore((state) => state.updateCachedDataset);
+    const updateCachedDataset = useSearchStore(
+        (state) => state.updateCachedDataset,
+    );
+
+    const fetchAllTimeTimeRange = useCallback(async (): Promise<
+        [Dayjs, Dayjs]
+    > => {
+        let sql: string;
+        if (CLP_STORAGE_ENGINES.CLP === SETTINGS_STORAGE_ENGINE) {
+            sql = buildClpTimeRangeSql();
+        } else {
+            sql = buildClpsTimeRangeSql(selectDataset ?? "default");
+        }
+        const resp = await querySql<
+            {
+                begin_timestamp: Nullable<number>;
+                end_timestamp: Nullable<number>;
+            }[]
+        >(sql);
+        const [timestamps] = resp.data;
+        if ("undefined" === typeof timestamps ||
+            null === timestamps.begin_timestamp ||
+            null === timestamps.end_timestamp
+        ) {
+            throw new Error();
```

Comment:
```suggestion
            throw new Error("Unable to get All Time range");
```
alternatively, use the `message` API from Antd to display an error message in the UI?

Reply:
sorry i just read the handling again and think we don't have to throw here. can we change the return type of fetchAllTimeTimeRange() to be `Promise<Nullable<[Dayjs, Dayjs]>>`, return null when the fields are not found, then the caller can just check for null instead of catching the error?